server/src-rsr/initialise.sql

```diff
@@ -280,11 +280,13 @@ CREATE TABLE hdb_catalog.event_log
   tries INTEGER NOT NULL DEFAULT 0,
   created_at TIMESTAMP DEFAULT NOW(),
   locked BOOLEAN NOT NULL DEFAULT FALSE,
-  next_retry_at TIMESTAMP
+  next_retry_at TIMESTAMP,
+  archived BOOLEAN NOT NULL DEFAULT FALSE
```

Comment:
Instead of doing this, would it be possible to add a foreign key reference to `event_triggers` with `ON DELETE SET NULL`? That seems like it would be essentially the same as adding this `archived` field, but with slightly more support from the database for data integrity.

Reply:
Hmm... it would remove the information of the trigger `name` though. I guess that is acceptable, except it would be a breaking change!

Reply:
> what if someone recreates an event trigger and expects the older events to be delivered as well.

To be honest, as a user, I think I would find this behavior extremely surprising… but I also don’t know that much about the use cases for event triggers. Are there other ways that users might currently observe the trigger name of events that have had their associated event trigger deleted? I guess I’m not 100% sure what the purpose of keeping the events around after the trigger has been deleted is in the first place, honestly.

Reply:
This was kinda an unfinished story. We had not finalized what to do with the event log of triggers that were deleted. We left it at the point of asking the user what they want explicitly: 

1) keep the events
2) delete the events
3) keep undelivered events till they are delivered

We don't have a statistic of how people are using event triggers and I don't mind doing 2) but it would be a breaking change which would need to be called out.

Reply:
Understood, that makes sense. I don’t know enough about the details here, so I guess the current approach seems alright for now. It does seem like it would be nice to have a more cohesive story for this in general, though.

Reply:
Agreed. Will open an issue for making this coherent.